pydicom/multival.py

```diff
@@ -38,39 +39,44 @@ class or type corresponding to the VR.
         """
         from pydicom.valuerep import DSfloat, DSdecimal, IS
 
-        self.type_constructor = type_constructor
+        def list_type_constructor(x):
+            return type_constructor(x) if x != '' else x
```

Comment:
Ok, done. @darcymason: please check the changed test for `DSfloat` etc. - as the argument is a class, not an instance, the previous check did not work, but maybe I didn't understand something.